js/ui/viewSelector.js

```diff
@@ -211,7 +211,9 @@ const ViewsDisplayLayout = new Lang.Class({
 
         let gridContainerBox = allocation.copy();
         gridContainerBox.y1 = this._computeGridContainerPlacement(gridContainerHeight, entryHeight, availHeight);
```

Comment:
I know this line is not part of this patch (it's my fault actually), but it would probably be nice to replace it with `gridContainerBox.y1 = heightAboveGrid` to prevent doing the same calculations twice